_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,324 @@
+---
+id: d0f69e
+name: All table header cells have assigned data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has assigned data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgements:
+  authors:
+    - Jey Nandakumar
+    - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that
+
+- has the [semantic role][] of [rowheader][] or [columnheader][] and;
+- is [visible][] and;
+- is [included in the accessibility tree][] and;
+- whose closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][];
+  1. has [descendants][descendant], that are non empty ("") and;
+  2. is [visible][] and;
+  3. is [included in the accessibility tree][].
+
+## Expectation
+
+Each target element is [assigned][] to at least one element with a [semantic role][] of [cell][] or [gridcell][].
+**Note:** The assigned cell is a [descendant][] in the [flat tree][] of the same [table][] or [grid][] element.
+
+## Assumptions
+
+This rule assumes that table header cells have a relationship conveyed through presentation with other cells within the same table.
+**Note:** This assumption helps exclude edge cases like: - a table definition where there is only one header cell, or - a table definition where there are multiple headers and no other cells
+
+## Accessibility Support
+
+Table markup and header cell association is not well supported by some popular assistive technologies. Passing this rule can still cause issues for users of those assistive technologies.
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [Forming relationships between data cells and header cells][assigned]
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The column header element has an [assigned][] cell, within the same `table` element.
```

Comment:
We kinda don't need the "within the same `table` element" part because it is implied by "assigned", and adding the precision makes it look like this is an extra condition rather than a consequence.
(same with other example descriptions)